src/GraphQL/Execution/QueryProcessor.php

```diff
@@ -275,6 +276,11 @@ protected function executeOperation(PromiseAdapter $adapter, ServerConfig $confi
 
       return $this->executeUncachableOperation($adapter, $config, $params, $document);
     }
+    catch (CacheableRequestError $exception) {
+      return $adapter->createFulfilled(
+        new QueryResult(NULL, [Error::createLocatedError($exception)], [], $exception)
+      );
+    }
     catch (RequestError $exception) {
       return $adapter->createFulfilled(new QueryResult(NULL, [Error::createLocatedError($exception)]));
```

Comment:
Can you, instead of adding another catch block, just add a ``$exception instanceof CachableMetadataInterace`` check here?

Reply:
@fubhy oh! sorry I didn't made the tests yet, but I saw this PR was merged.
Why do you think we should have the instanceof check instead of a catch block?